src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,98 @@
 import { useState } from 'react';
+
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+
+type NewMovieProps = {
+  onAdd: (newMovie: Movie) => void;
+};
 
-export const NewMovie = () => {
+export const NewMovie = ({ onAdd }: NewMovieProps) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [count, setCount] = useState(0);
+
+  const isFormValid = Boolean(
+    title.trim() !== '' &&
+      imgUrl.trim() !== '' &&
+      imdbUrl.trim() !== '' &&
+      imdbId.trim() !== '',
```

Comment:
```suggestion
    title.trim() &&
      imgUrl.trim() &&
      imdbUrl.trim() &&
      imdbId.trim(),
```
You don't need to compare with an empty string, just check if value exists
https://developer.mozilla.org/en-US/docs/Glossary/Truthy